src/main/java/com/patchworkmc/util/LambdaUtil.java

```diff
@@ -0,0 +1,71 @@
+package com.patchworkmc.util;
+
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+import org.objectweb.asm.Handle;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+
+public final class LambdaUtil {
+	private LambdaUtil() {
+	}
+
+	public static final Handle METAFACTORY = new Handle(Opcodes.H_INVOKESTATIC, "java/lang/invoke/LambdaMetafactory", "metafactory",
+			"(Ljava/lang/invoke/MethodHandles$Lookup;Ljava/lang/String;Ljava/lang/invoke/MethodType;Ljava/lang/invoke/MethodType;Ljava/lang/invoke/MethodHandle;Ljava/lang/invoke/MethodType;)Ljava/lang/invoke/CallSite;",
+			false);
+	public static final Type OBJECT_METHOD_TYPE = Type.getMethodType("(Ljava/lang/Object;)V");
+	public static final Type DUAL_OBJECT_METHOD_TYPE = Type.getMethodType("(Ljava/lang/Object;Ljava/lang/Object;)V");
+
+	// TODO: Does this work on non-reference lambdas?
+	/**
+	 * Generates an {@code INVOKEDYNAMIC} instruction for a {@link Consumer} that is a reference to an instance method.
+	 * <p>
+	 * On the stack, this will pop the instance (which should be of type {@code className}) and push a {@link Consumer}.
+	 * </p>
+	 *
+	 * @param visitor the {@link MethodVisitor} for the {@code INVOKEDYNAMIC} instruction to be written to.
+	 * @param callingOpcode the opcode needed to invoke the method. The calling opcode *must* be a {@code H_INVOKE*} instruction.
+	 * @param className the class that holds the target method
+	 * @param methodName the target method's name
+	 * @param methodDescriptor the target method's descriptor
+	 * @param isInterface whether the target class is an interface or not
+	 */
+	public static void visitConsumerInstanceMethodReference(MethodVisitor visitor, int callingOpcode, String className, String methodName, String methodDescriptor, boolean isInterface) {
```

Comment:
These method names are very verbose, is there a way to trim them down?

Reply:
I swapped "methodReference" with "lambda"